### DISCUSSION TITLE
Is there a way to display image alt text as caption?

### DISCUSSION THREAD

**Discussion by tisonkun** [facebook/docusaurus#8667] [Q&A] [answered]
For example, I'd like to display:

```md
![Neofetch 时刻](...)
```

as:

<img width="916" alt="image" src="https://user-images.githubusercontent.com/18818196/218925971-127a864f-8079-4507-a40e-5214bf1f9425.png">

Maybe https://fancyapps.com/fancybox can help here.

**Comment by Josh-Cena** [comment, marked as answer]
You can make a [Remark plugin](https://docusaurus.io/docs/markdown-features/plugins) to walk all `image` nodes and transform them to use `<figure>` and `<figcaption>`. Here's an example:

```js
const visit = require("unist-util-visit");

const plugin = (options) => {
  const transformer = async (root) => {
    visit(root, "paragraph", (node) => {
      if (node.children.length === 1 && node.children[0].type === "image") {
        const { url, alt } = node.children[0];
        if (url.includes("youtu.be") || url.includes("youtube.com") || !alt) {
          return;
        }
        for (const member in node) {
          delete node[member];
        }
        node.type = "jsx";
        node.value = `<figure>
  <img src="${url}" alt={"${alt.replaceAll(
          '"',
          '\\"'
        )}"} style={{ marginBottom: 10 }} />
  <figcaption style={{ textAlign: "center", color: "gray", fontSize: 14 }}>
    ${alt}
  </figcaption>
</figure>`;
      }
    });
  };
  return transformer;
};

module.exports = plugin;
```

However, I strongly discourage you to use alt messages as captions. Alt messages should be much, much longer than the caption, because it is a direct substitution for what the image itself contains. If assistive technologies is not relatable enough, think about when your internet is down—you will see a broken image with the alt text, then the caption with duplicated text. This is not good. See https://github.com/facebook/docusaurus/pull/8047 for an example of writing accessible alt messages.

**Reply by tisonkun** [reply]
Then it reads to me that I should define a custom component to fulfill alt text and caption separately?

**Reply by Josh-Cena** [reply]
Exactly. Alternatively you can use some custom Markdown, let's say:

```md
![alt](src "title") -- caption
```

This would produce a paragraph with an image and a text as its children, and you could parse that instead.

**Comment by enthusiasticfolk** [comment]
Why do you want to do that? You often optimize your alts, and that optimization will look in the caption. Won't it?
